Source/Project64/N64 System/Mips/Pif Ram.cpp

```diff
@@ -297,19 +297,16 @@ void CPifRam::SI_DMA_READ()
 			{
 				continue;
 			}
-			RDRAM[RdramPos ^3] = m_PifRam[count];
+			RDRAM[BES(RdramPos)] = m_PifRam[count];
 		}
 	}
 	else
 	{
-		for (size_t i = 0; i < 64; i += 4)
-		{
-			unsigned __int32 pif_ram_dword;
-			std::memcpy(&pif_ram_dword, &PifRamPos[i], sizeof(unsigned __int32));
-
-			pif_ram_dword = swap32by8(pif_ram_dword);
+		register size_t i;

```

Comment:
>There's no particular reason to leave this in an outer scope it's not used in.

That assumes the view that "scope" should exist in between.

If you suppose that in between the body of a loop (such as `for`) and just outside the loop, should be a scope of its own (the control and heading of the loop), then it is your opinion that the declaration should go to the loop header.

In particular C99 and C++ should allow just as you say, but I actually find that the C89 version looks less open to personal interpretation and less ambiguous, IMO, and easier for me to read in my honest opinion, if not also easier for the compiler to parse faster.  (Said issues with direct language parse-ability to compilers having constituted many of the other hold-backs of optional features not present before C99.)

But if it looks just that much harder for you or @project64 to read, I will change it back.

>By the way, have you benchmarked or checked that the assembly produced is actually better with the register keyword.  I can bet it does not alter semantics at all

Then you propose that my purpose for specifying `register` is that I think otherwise?

Like any hint for any language, I'm aware that it makes no automatic difference in the end result.  It is just a compiler hint--one which often goes ignored (especially in this case, as the compiler's job is to optimize which should be registers, which should be on the stack, and which should be in-lined or even exist at all).  (And it **usually** has all the information it needs to correctly decide such.)

But unless you can find a reason to suggest that there's anything to lose from my approach, and not just that "there's _almost_ never anything to gain", then I'm really not sure what positive difference it makes to change that.

Another minor use for `register` is that it prevents taking the address of something by mistake.